diff --git a/packages/drive-store/store/_links/extendedAttributes.ts b/packages/drive-store/store/_links/extendedAttributes.ts
index 7128349883..dc285a7377 100644
--- a/packages/drive-store/store/_links/extendedAttributes.ts
+++ b/packages/drive-store/store/_links/extendedAttributes.ts
@@ -193,7 +193,7 @@ export function createDocumentExtendedAttributes() {
     };
 }
 
-async function encryptExtendedAttributes(
+export async function encryptExtendedAttributes(
     xattr: ExtendedAttributes,
     nodePrivateKey: PrivateKeyReference,
     addressPrivateKey: PrivateKeyReference
