diff --git a/packages/drive-store/store/_links/useLinksState.tsx b/packages/drive-store/store/_links/useLinksState.tsx
index a6ac0b3fb2..2c287e8203 100644
--- a/packages/drive-store/store/_links/useLinksState.tsx
+++ b/packages/drive-store/store/_links/useLinksState.tsx
@@ -48,6 +48,13 @@ export function useLinksStateProvider() {
 
     const [state, setState] = useState<LinksState>({});
 
+    const removeLinkForDriveCompat = useCallback(
+        (shareId: string, linkId: string) => {
+            setState((state) => deleteLinks(state, shareId, [linkId]));
+        },
+        [state]
+    );
+
     useEffect(() => {
         const callbackId = eventsManager.eventHandlers.register((_volumeId, events, processedEventCounter) =>
             setState((state) => updateByEvents(state, events, processedEventCounter))
@@ -165,6 +172,8 @@ export function useLinksStateProvider() {
         getAllShareLinks, // This should be use only in specific case when you know the links you need (Ex: Bookmarks)
         removeLinkForMigration,
         removeLinkForSharedWithMe,
+        /** Should never be used outside of `drive-store`. */
+        removeLinkForDriveCompat,
     };
 }
 
